### ISSUE TITLE
Tensorboard could not bind to unsupported address family ::

### ISSUE THREAD

**Issue by oe3gwu** [determined-ai/determined#7814]
### Describe the bug

Using an HPE Ezmeral Stack, we tried to use Tensorboard within determined.ai. Determined works great but we cant start Tensorboard. It seems it tries to create an IPv6 connection, which our system doesnt support, and we cant enable it. IPv4 connection should be enough for tensorboard.

### Reproduction Steps

1. given finished experiment, using the web interface, start a tensorboard
2. in the newly opened tab, on the top of the window open the logs
3. view the logs (attached)
4. the tensorboard pod crashes immediatly.

### Expected Behavior

tensorboard starts without issues and only using ipv4

### Screenshot

none, please see additional context for logs.

### Environment

- Device or hardware: HPE Apollo 6500 with 4x NVidia A100 HGX
- OS: Redhat 8 on worker nodes
- Browser: Chrome, Edge, Firefox
- Version: latest

If you need more info, feel free to contact me.

### Additional Context

[determined-tensorboard.txt](https://github.com/determined-ai/determined/files/12548319/determined-tensorboard.txt)

As stated in the logs, tensorboard during startup tries to connect to localhost using ipv6 (::). We assume it has something to do with the proxying, but the issue clearly is part of the tensorboard application. We assume this is the issue, because our stack isnt able to support ipv6, and we had similar issues with other applications in the past, but there we were able to manually disable ipv6 via a flag in the helm chart (values.yaml).

**Comment by rb-determined-ai** [comment]
What does it mean that your system doesn't support ivp6?  Like, if I wanted to set up my system in the same way to reproduce what you are seeing, how would I do it?

Additionally, in the mean time, we can check one possible fix on your system.  Could you write a `startup-hook.sh` file with the following contents:

```
file="$(python -c 'import determined.exec.tensorboard as x; print(x.__file__)')"
sed -i -e 's/tensorboard_args.append.*bind_all.*/pass/' "$file"
```

And then run the following command:

```
det tensorboard start --include startup-hook.sh 1
```

Then share the logs from that tensorboard task with me.

**Comment by oe3gwu** [comment]
We will try. I get back to you at least on monday, because I am not sure if I am on the office tomorrow. Thank you!

**Comment by N520** [comment]
Thanks for the input! We reran the tensorboard with the given `startup-hook.sh` as you said and now the tensorboard does not immediately crash. The logs are attached.
[tensorboard_running_log.txt](https://github.com/determined-ai/determined/files/12574709/tensorboard_running_log.txt)


But  the tensorboard browser page only shows a blank white page and the console prints the following 502 error:
`GET <DETERMINED_MASTER>/proxy/f212f38a-892d-48af-bb97-8f792f3c5eaa/ 502 (Bad Gateway)`

Are we missing some proxy configuration on the determined master?

**Comment by oe3gwu** [comment]
N520 is workmate of me. We talk about the same issue! Just FYI! :-)

**Comment by rb-determined-ai** [comment]
> What does it mean that your system doesn't support ivp6? Like, if I wanted to set up my system in the same way to reproduce what you are seeing, how would I do it?

Sorry, I still need the answer to the above question.

> But the tensorboard browser page only shows a blank white page and the console prints the following 502 error

I wouldn't expect the master to work without that `--bind_all` flag that we removed from the command line.  I think tensorboard binds to 127.0.0.1 with it.  `--bind_all` was the source of the attempt to bind to `::`, that's what we showed.  As another test, you could configure `--host 0.0.0.0`.  You'll need to do it with the `tensorboard_args` in a yaml config file, as described in our [docs](https://docs.determined.ai/latest/interfaces/tensorboard.html#customizing-tensorboard-instances).  Something like this:

```yaml
tensorboard_args: ["--host", "0.0.0.0"]
```

**Comment by oe3gwu** [comment]
Regarding IPv6, our cluster doesnt provide an enabled IPv6 stack. Not on the physical interfaces, and not on the virtual one inside Kubernetes. That means if :: is tried to bind (the IPv6 equivalent of localhost, I think), it will fail, because the worker doesnt know the address format :: . The problem is that kubernetes is stupid enough that if it wants IPv4 and IPv6 dual stack, that there is a chance to fail in the execution if a non existent IPv6 stack creates an error message. Kubecost got a similar issue, for example.

In short, the command `ip a` only list IPv4. No IPv6 link-local or anything else.

We will try your solution and come back to you.

**Comment by rb-determined-ai** [comment]
any updates?

**Comment by rb-determined-ai** [comment]
closing due to inactivity
